source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -0,0 +1,183 @@
+
+            if (min_value_for_particle_property > particle_property_value)
+              min_value_for_particle_property = particle_property_value;
+            if (max_value_for_particle_property < particle_property_value)
+              max_value_for_particle_property = particle_property_value;
+          }
+
+        dealii::LAPACKFullMatrix<double> B(matrix_dimension, matrix_dimension);
+        dealii::LAPACKFullMatrix<double> B_inverse(B);
+
+        Vector<double> c_ATr(matrix_dimension);
+        Vector<double> c(matrix_dimension);
+
+        const double threshold = 1e-15;
+        unsigned int index_positions = 0;
+
+        /**
+         * Matrix A can be rank deficient if it does not have full rank, therefore singular.
+         * To circumvent this issue, we solve A^TAx=A^Tr by using singular value
+         * decomposition (SVD).
+         */
+        A.Tmmult(B, A, false);
+        A.Tvmult(c_ATr,r);
+        B_inverse.compute_svd();
+        B_inverse.compute_inverse_svd(threshold);
+        B_inverse.vmult(c, c_ATr);
+
+        for (typename std::vector<Point<dim> >::const_iterator itr = positions.begin(); itr != positions.end(); ++itr, ++index_positions)
+          {
+            const Point<dim> support_point = *itr;
+            // TODO: Removed const
+            double interpolated_value = c[0] +
+                                        c[1]*(support_point[0] - approximated_cell_midpoint[0])/cell_diameter +
+                                        c[2]*(support_point[1] - approximated_cell_midpoint[1])/cell_diameter +
+                                        c[3]*(support_point[0] - approximated_cell_midpoint[0])*(support_point[1] - approximated_cell_midpoint[1])/std::pow(cell_diameter,2);
+
+            cell_properties[index_positions][property_index] = interpolated_value;
+
+            if (interpolated_value > max_value_for_particle_property)
+              interpolated_value = max_value_for_particle_property;
+            else if (interpolated_value < min_value_for_particle_property)
+              interpolated_value = min_value_for_particle_property;
+          }
+        return cell_properties;
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Interpolator
+    {
+      ASPECT_REGISTER_PARTICLE_INTERPOLATOR(BilinearLeastSquares,
+                                            "bilinear least squares",
+                                            "Interpolates particle properties onto a vector of points using a "
+                                            "bilinear least squares method. Currently only 2D models are "
+                                            "supported. Note that because no limiter is utilized this may "
+                                            "result in overshoot and/or undershoot of interpolated property.")
```

Comment:
can you add a note that this feature requires deal.II configured with blas/lapack?